### ISSUE TITLE
Hand Crafted Prompt design

### ISSUE THREAD

**Issue by Thamirawaran** [Jaseci-Labs/vlm-research#5]
Design different hand crafted prompts

Consider these things while designing prompts
1. Give context, task instruction: These are car damage images. you are going to write insurance report by indentifying damages
2. Give sample output example
3. Give guidance on what should be retained throughout the outputs
4. You may add special tokens to manipulate the model
5. If you have any other ideas consider them also

Comeup with more than 5 prompts. 

After the experiment, analyse the results and select 2 or 3.

Research papers: 
[A Good Prompt Is Worth Millions of Parameters](https://arxiv.org/abs/2110.08484)

**End goal:** This type(.....) of prompt is working better. Reason is ............ How it can be applied to other tasks. How can we add special tokens i a prompt

**Comment by Gayanukaa** [comment]
Prompts required categories
- example prompt (output in this format sample answer)
- complexity
- roleplay

**Comment by Gayanukaa** [comment]
Reference document: [google doc tab](https://docs.google.com/document/d/1WYQq5K3UsR7GcwK49T6jjMCzu4cMXcfRR20pya6VsiU/edit?tab=t.cng2vdwjdh0l)

**Comment by Gayanukaa** [comment]
### 📌 Reference from Paper ([2110.08484v2](https://arxiv.org/abs/2110.08484))

#### 1) **No prompt (baseline with no guidance)**  
- Zero-shot results are remarkably affected by input prompts, with "no prompt" performing significantly worse than hand-crafted prompts.  
- With larger training data in the few-shot setting, "no prompt" can sometimes perform similarly to or better than some noisy or even hand-crafted prompts.

#### 2) **Noisy prompt (irrelevant prompts, noisy tokens, and random sentences)** *(Table 11)*  
- Understand the effect of misleading or irrelevant guidance.  
-  *Irrelevant prompts*: Random questions or instructions that are unrelated to the task.  
- *Noisy tokens*: Randomly selected tokens from the model's vocabulary.  
-  *Random sentences*: Captions from a different dataset (MS COCO) which provide incorrect information for the given image.

#### 3) **An image of….**  
- Hand-crafted descriptive prompt to check zero-shot results.

#### 4) **Descriptive prompt with stylistic guidance and longer length**  
- A basic structured prompt like `"Question: [Q] Answer:"` (P2) to guide clarity, length, and format of generated answers.

#### 5) **Masked prompt (there is a <> in this image. it does <>)**  
- Hand-crafted prompts that incorporate specific structural elements or tokens to guide the model, often mimicking the format of the pre-training objectives.  
- Use of sentinel tokens, such as `<text_1>`, which are similar to those used in MaskedLM pre-training.
